chanfitness/revenue.go

```diff
@@ -0,0 +1,52 @@
+package chanfitness
+
+import (
+	"github.com/lightningnetwork/lnd/channeldb"
+	"github.com/lightningnetwork/lnd/lnwire"
+)
+
+// attributeIncoming is the percentage of fees that are attributed to the
+// incoming channel in a forwarding event. The remaining 1- attributeIncoming
+// percent is attributed to the outgoing channel.
+var attributeIncoming = 0.5
+
+// calculateRevenue calculates the revenue for a channel for the set of events
+// provided. It distributes fees between incoming and outgoing channels
+// involved in htlc forwards according the the value set in attributeIncoming.
+func calculateRevenue(chanID lnwire.ShortChannelID,
+	events []channeldb.ForwardingEvent) lnwire.MilliSatoshi {
+
+	// Revenue is the total revenue earned by this channel in millisatoshis.
+	// This value is accumulated as a float then converted to millisatoshis so
+	// that we can minimize the amount of rounding from float -> int.
+	var revenue float64
+
+	for _, event := range events {
+		// Calculate fees earned for this event.
+		fee := event.AmtIn - event.AmtOut
+
+		// If our target channel was the incoming channel for this event,
+		// attribute the incomingAttribute percentage of the fees to it.
+		if event.IncomingChanID == chanID {
+			earned := float64(fee) * attributeIncoming
+
+			log.Debugf("Channel: %v is outgoing channel, incrementing "+
+				"revenue by: %v msat", chanID, earned)
+
+			revenue += earned
+		}
+
+		// If our target channel was the outgoing channel for this event,
+		// attribute (1 - percentage for incoming) percent of the fees to it.
+		if event.OutgoingChanID == chanID {
+			earned := float64(fee) * (1 - attributeIncoming)
+
+			log.Debugf("Channel: %v is outgoing channel, incrementing "+
```

Comment:
Perhaps trace instead? Otherwise this could be rather spammy for a popular or long lived node.